src/cplus_plugin/gui/constant_raster_manager_dialog.py

```diff
@@ -0,0 +1,1135 @@
+        self.lst_pathways.selectionModel().selectionChanged.connect(
+            self._on_model_component_selection_changed
+        )
+
+        self._activities_model = ActivityItemModel(
+            load_pathways=False, is_checkable=True
+        )
+        self.lst_activities.setModel(self._activities_model)
+        for activity in settings_manager.get_all_activities():
+            self._activities_model.add_activity(activity)
+
+        # Connect to the view's selection model, not the model itself
+        self.lst_activities.selectionModel().selectionChanged.connect(
+            self._on_model_component_selection_changed
+        )
+
+        # Connections
+        self.cbo_raster_type.currentIndexChanged.connect(
+            self.on_raster_type_selection_changed
+        )
+        self.rb_pathway.toggled.connect(self.on_pathway_type_selected)
+        self.rb_activity.toggled.connect(self.on_activity_type_selected)
+        # Note: btn_create_current uses menu actions, not direct clicked connection
+        self.btn_close.clicked.connect(self.close)
+
+        # Connect model itemChanged signals to save checkbox states
+        self._pathways_model.itemChanged.connect(self._on_item_checked_changed)
+        self._activities_model.itemChanged.connect(self._on_item_checked_changed)
+
+        # Connect output range spinboxes to save the values to the collection
+        self.spin_min_value.valueChanged.connect(self.on_output_range_changed)
+        self.spin_max_value.valueChanged.connect(self.on_output_range_changed)
+
+        # Restore UI state from saved components
+        self._restore_ui_state()
+
+    def update_component_type_labels(self, label: str):
+        """Update UI labels based on component type.
+
+        :param label: Either "NCS Pathways" or "Activities"
+        """
+        if label == "NCS Pathways":
+            self.sw_model_type_container.setCurrentIndex(0)
+            self._left_group.setTitle("NCS Pathways")
+            self.lbl_description.setText(
+                "<b>Description:</b> Constant rasters for NCS pathways"
+            )
+        elif label == "Activities":
+            self.sw_model_type_container.setCurrentIndex(1)
+            self._left_group.setTitle("Activities")
+            self.lbl_description.setText(
+                "<b>Description:</b> Constant rasters for activities"
+            )
+
+    def load_pathways(self):
+        """Load pathway metadata into the raster type dropdown and restore selection."""
+        # Get metadata for pathways
+        pathway_metadatas = self.constant_raster_registry.metadata_by_component_type(
+            ModelComponentType.NCS_PATHWAY
+        )
+        self.cbo_raster_type.clear()
+        for metadata in pathway_metadatas:
+            self.cbo_raster_type.addItem(metadata.display_name, metadata.id)
+
+        # Try to restore the previously selected pathway raster type
+        settings = QgsSettings()
```

Comment:
@osundwajeff all settings-related functionality should be defined in `conf.py` for consistency and re-usability.